lib/sles4sap/sap_deployment_automation_framework/configure_tfvars.pm

```diff
@@ -0,0 +1,121 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::configure_tfvars;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use utils qw(file_content_replace);
+use sles4sap::azure_cli qw(az_storage_blob_list);
+use sles4sap::sap_deployment_automation_framework::deployment qw(get_os_variable);
+use sles4sap::sap_deployment_automation_framework::deployment_connector qw(find_deployment_id);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation that help with preparation of tfvars file.
+
+=cut
+
+our @EXPORT = qw(
+  prepare_tfvars_file
+);
+
+=head2 prepare_tfvars_file
+
+    prepare_tfvars_file(deployment_type=>$deployment_type);
+
+=over 1
+
+=item B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+=back
+
+Downloads tfvars template files from openQA data dir and places them into correct place within SDAF repo structure.
+Returns full path of the tfvars file.
+
+=cut
+
+sub prepare_tfvars_file {
+    my (%args) = @_;
+    croak 'Deployment type not specified' unless $args{deployment_type};
+    my %tfvars_os_variable = (
+        deployer => 'deployer_parameter_file',
+        sap_system => 'sap_system_parameter_file',
+        workload_zone => 'workload_zone_parameter_file',
+        library => 'library_parameter_file'
+    );
+    croak "Unknown deployment type: $args{deployment_type}" unless $tfvars_os_variable{$args{deployment_type}};
+
+    my %tfvars_template_url = (
+        deployer => data_url('sles4sap/sap_deployment_automation_framework/DEPLOYER.tfvars'),
+        sap_system => data_url('sles4sap/sap_deployment_automation_framework/SAP_SYSTEM.tfvars'),
+        workload_zone => data_url('sles4sap/sap_deployment_automation_framework/WORKLOAD_ZONE.tfvars'),
+        library => data_url('sles4sap/sap_deployment_automation_framework/LIBRARY.tfvars')
+    );
```

Comment:
A bit of Perl's crazyness suggestion:
```
my @valid_deployments = qw(deployer sap_system workload_zone library);
croak "Unknown deployment type: $args{deployment_type}" unless (grep /$args{deployment_type/ @valid_deployments);

my %tfvars_os_variable = map { $_ => "${_}_parameter_file" } @valid_deployments;
my %tfvars_template_url = map { $_ => data_url('sles4sap/sap_deployment_automation_framework/' . uc($_) . '.tfvars') } @valid_deployments;
```

Reply:
As much as I like maps, I am inclined to keep more readable version here.
 
In any case... I have a very secret and very evil plan to quietly and slowly remove templates and generate full tfvars inside testcode. So those lines will just go away. :D
Don't tell anyone.